src/main/kotlin/org/rust/ide/refactoring/extractFunction/ExtractFunctionParameterTablePanel.kt

```diff
@@ -17,6 +23,57 @@ class ParameterDataHolder(val parameter: Parameter, val onChange: () -> Unit) :
     }
 }
 
+class ChooseColumn : ColumnInfo<ParameterDataHolder, JBCheckBox>(null) {
+
+    override fun valueOf(item: ParameterDataHolder): JBCheckBox? = JBCheckBox(null, item.parameter.isSelected).apply {
+        this.horizontalAlignment = JBCheckBox.CENTER
+        this.addChangeListener {
+            item.parameter.isSelected = this.isSelected
+            item.onChange()
+        }
+    }
+
+    override fun getColumnClass(): Class<*> = JBCheckBox::class.java
+
+    override fun isCellEditable(item: ParameterDataHolder?): Boolean = true
+}
+
+class ChooseRenderer : TableCellRenderer {
+
+    override fun getTableCellRendererComponent(
+        table: JTable,
+        value: Any?,
+        isSelected: Boolean,
+        hasFocus: Boolean,
+        row: Int,
+        column: Int
+    ): Component {
+        table.columnModel.getColumn(column).run {
+            maxWidth = WIDTH
+            preferredWidth = WIDTH
+        }
+
+        return (value as JBCheckBox).apply { background = table.background }
+    }
+
+    companion object{
+        private const val WIDTH = 40
+    }
+}
+
+class ChooseEditor(private var isSelected: Boolean = false) : AbstractCellEditor(), TableCellEditor {
```

Comment:
Looks like you can use `BooleanTableCellEditor` instead of introducing your own editor